Rebus.AdoNet/Dialects/TypeNames.cs

```diff
@@ -1,176 +1,182 @@
-		{
-			type = ReplaceOnce(type, LengthPlaceHolder, size.ToString());
-			type = ReplaceOnce(type, ScalePlaceHolder, scale.ToString());
-			return ReplaceOnce(type, PrecisionPlaceHolder, precision.ToString());
-		}
-
-		/// <summary>
-		/// Get default type name for specified type
-		/// </summary>
-		/// <param name="typecode">the type key</param>
-		/// <returns>the default type name associated with the specified key</returns>
-		public string Get(DbType typecode)
-		{
-			string result;
-			if (!defaults.TryGetValue(typecode, out result))
-			{
-				throw new ArgumentException("Dialect does not support DbType." + typecode, "typecode");
-			}
-			return result;
-		}
-
-		/// <summary>
-		/// Get the type name specified type and size
-		/// </summary>
-		/// <param name="typecode">the type key</param>
-		/// <param name="size">the SQL length </param>
-		/// <param name="scale">the SQL scale </param>
-		/// <param name="precision">the SQL precision </param>
-		/// <returns>
-		/// The associated name with smallest capacity >= size if available and the
-		/// default type name otherwise
-		/// </returns>
-		public string Get(DbType typecode, uint size, uint precision, uint scale)
-		{
-			SortedList<uint, string> map;
-			weighted.TryGetValue(typecode, out map);
-			if (map != null && map.Count > 0)
-			{
-				foreach (KeyValuePair<uint, string> entry in map)
-				{
-					if (size <= entry.Key)
-					{
-						return Replace(entry.Value, size, precision, scale);
-					}
-				}
-			}
-			//Could not find a specific type for the size, using the default
-			return Replace(Get(typecode), size, precision, scale);
-		}
-
-		/// <summary>
-		/// For types with a simple length, this method returns the definition
-		/// for the longest registered type.
-		/// </summary>
-		/// <param name="typecode"></param>
-		/// <returns></returns>
-		public string GetLongest(DbType typecode)
-		{
-			SortedList<uint, string> map;
-			weighted.TryGetValue(typecode, out map);
-
-			if (map != null && map.Count > 0)
-				return Replace(map.Values[map.Count - 1], map.Keys[map.Count - 1], 0, 0);
-
-			return Get(typecode);
-		}
-
-		/// <summary>
-		/// Set a type name for specified type key and capacity
-		/// </summary>
-		/// <param name="typecode">the type key</param>
-		/// <param name="capacity">the (maximum) type size/length</param>
-		/// <param name="value">The associated name</param>
-		public void Put(DbType typecode, uint capacity, string value)
-		{
-			SortedList<uint, string> map;
-			if (!weighted.TryGetValue(typecode, out map))
-			{
-				// add new ordered map
-				weighted[typecode] = map = new SortedList<uint, string>();
-			}
-			map[capacity] = value;
-		}
-
-		/// <summary>
-		/// 
-		/// </summary>
-		/// <param name="typecode"></param>
-		/// <param name="value"></param>
-		public void Put(DbType typecode, string value)
-		{
-			defaults[typecode] = value;
-		}
-	}
+﻿using System;
```

Comment:
Please, this file appears to have changed entirely. Fix it so only changed parts appear on this diff.

Reply:
ack